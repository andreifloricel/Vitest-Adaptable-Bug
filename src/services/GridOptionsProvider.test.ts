import { spyOn } from 'vitest';
import GridOptionsProvider from './GridOptionsProvider';
import { columnDefinitionsProvider } from './ColumnDefintionsProvider';

describe('GridOptionsProvider', () => {
    it('should provide grid options', () => {
        const columnDefinitionsProviderSpy = spyOn(columnDefinitionsProvider, 'getColumnDefinitions');
        columnDefinitionsProviderSpy.mockReturnValue([]);
        const sut = new GridOptionsProvider();
        const result = sut.getGridOptions();
        expect(result).toMatchSnapshot();
    });
});
