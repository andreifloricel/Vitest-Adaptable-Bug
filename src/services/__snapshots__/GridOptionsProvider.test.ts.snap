// Vitest Snapshot v1

exports[`GridOptionsProvider > should provide grid options 1`] = `
{
  "columnDefs": [],
  "components": {
    "AdaptableToolPanel": [Function],
  },
  "enableRangeSelection": true,
  "sideBar": true,
  "suppressMenuHide": true,
}
`;

exports[`GridOptionsProvider should provide grid options 1`] = `
Object {
  "columnDefs": Array [],
  "components": Object {
    "AdaptableToolPanel": [Function],
  },
  "enableRangeSelection": true,
  "sideBar": true,
  "suppressMenuHide": true,
}
`;
